### ISSUE TITLE
TestError_SystemErrno fails with "The segment is already unlocked"

### ISSUE THREAD

**Issue by elagergren-spideroak** [mattn/go-sqlite3#786]
Our CI is showing this error when we run the unit tests:

```
build	12-Feb-2020 19:50:14	--- FAIL: TestError_SystemErrno (0.00s)
build	12-Feb-2020 19:50:14	    error_test.go:273: expected SystemErrno to be a not exists error, but got The segment is already unlocked.
```

OS: Windows 10.0.17763.1
Go: 1.13.4

**Comment by rittneje** [comment]
That error code is returned by the operating system. It sounds like there may be some issue with your Windows installation? Unfortunately, I cannot turn up much information. https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-#error_not_locked

If it continues to happen, I would suggest posting to the SQLite mailing list to see if anyone there can shed some light on the matter. https://www.sqlite.org/support.html

**Comment by elagergren-spideroak** [comment]
I think, for some reason, we also did not update the amalgamation. Apologies for the noise.

**Comment by elagergren-spideroak** [comment]
Reopening as this still occurs after updating the amalgamation.

**Comment by rittneje** [comment]
@elagergren-spideroak That error code is retrieved by calling the `sqlite3_system_errno` function, documented [here](https://www.sqlite.org/c3ref/system_errno.html).  Like I said, you need to post on the SQLite mailing list to see if they can explain why this is happening. That error is being reported by the operating system itself for some reason, so perhaps there is some issue with the Windows VFS.

**Comment by ericlagergren** [comment]
@rittneje yup, I am going to do that. I’m just making sure it’s well-documented in case anybody else runs into it. 👍
